### PR TITLE
Planar projection camera

### PR DESCRIPTION
**Pull request by thatcomputerguy0101**
Revision of #42 to not depend on the zooming functionality of #40. Keeping the near and far planes on the correct side of the focal point is tricky with this logic, resulting in the additional check and adjustment in `set_planar_projection`. Reimplementing this projection using the `Viewer` trait externally would allow for the alternative zoom logic to make that check unnecessary. However, that solution would make the planar projection incompatible with the built-in control samples.

### CODE REVIEW

File: src/camera.rs [hunk 1]
```diff
@@ -563,6 +638,7 @@ impl Camera {
     fn screen2ray(&self) -> Mat4 {
         let mut v = self.view;
         if let ProjectionType::Perspective { .. } = self.projection_type {
+            v /= v[3][3];
```
Comment (asny) [inline src/camera.rs:641]: Why is this needed? 🤔

**Review by asny — COMMENTED** [review]
Looks good except the change in `screen2ray` 👇

**Comment by thatcomputerguy0101** [comment]
To resolve the problem that led to the focal point check, maybe the near and far plane distances should be marked as either absolute or relative, where absolute is the current behavior and relative gets multiplied by the zoom factor (matching how the old zoom refactor effectively handled the plane distances). This reduces the complexity of the planar projection by making the relative mode the recommended mode, but increases the complexity of the perspective projection which would need to be recalculated if either depth uses relative mode.

**Comment by thatcomputerguy0101** [comment]
Regarding the absolute/relative (world/camera?) distance units, does it make more sense to force `z_near` and `z_far` to share the same unit, or should the units be individually specified?

**Comment by asny** [comment]
I think this is fine, I still think using a planar camera is a bit of an edge case, so I doubt if a lot of people care. I think the check in `set_planar_projection` is not super nice, but it's also not a huge issue IMO. I think it's more important to keep it simple and still use absolute values for `z_near` and `z_far`.
